project/src/ExternalInterface.cpp

```diff
@@ -1126,7 +1133,10 @@ namespace lime {
 
 		#ifdef LIME_FREETYPE
 		Font *font = (Font*)val_data (fontHandle);
-		return font->GetGlyphIndex ((char*)character.c_str ());
+		const std::string str (hxs_utf8 (character, nullptr));
+		const std::unique_ptr<char[]> buff (new char[str.size() + 1]);
+		std::memcpy (buff.get (), str.c_str (), str.size () + 1);
+		return font->GetGlyphIndex (buff.get ());
```

Comment:
Yes, that makes sense. As far as I can tell, `GetGlyphIndex` and `readNextChar` (which `GetGlyphIndex` calls) don't actually modify the string, so I think it would be good to change the function signatures so they take in const arguments, which would allow us to avoid this buffer, and it would also make it more clear that the function only reads from the string.

Reply:
There are certainly some things that could use some care too but I wanted to keep this PR as sort as possible in order to increase the chances for it to be accepted. I really didn't expect to take 2 years before someone even pays any attention to it.

Reply:
I guess we can deal with const issue later: #1671